File: src/test/resources/TransposeMatrix.java
public class Solution {

    public void transposeMatrix(int[][] matrix) {
        int n = matrix.length - 1;
        for (int i = 0; i <= n; i++) {
            for (int j = i + 1; j <= n; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
    }
}

File: src/test/resources/TransposeMatrix.kt
class TransposeMatrix {

    fun transposeMatrix(matrix: Array<IntArray>) {
        val n = matrix.size - 1
        for (i in 0..n) {
            for (j in i + 1..n) {
                val temp = matrix[i][j]
                matrix[i][j] = matrix[j][i]
                matrix[j][i] = temp
            }
        }
    }
}
